rules/typescript.js

```diff
@@ -0,0 +1,62 @@
+const typescriptRules = {
+  '@typescript-eslint/adjacent-overload-signatures': 2,
+  '@typescript-eslint/array-type': 2,
+  '@typescript-eslint/await-thenable': 2,
+  '@typescript-eslint/ban-types': [
+    2,
+    {
+      types: {
+        Array: null,
+        Object: 'Use {} instead',
+        String: {
+          message: 'Use string instead',
+          fixWith: 'string',
+        },
+      },
+    },
+  ],
+  '@typescript-eslint/camelcase': [2, { properties: 'never' }],
+  '@typescript-eslint/class-name-casing': 2,
+  '@typescript-eslint/explicit-function-return-type': 0,
+  '@typescript-eslint/explicit-member-accessibility': 0,
+  '@typescript-eslint/indent': 0,
+  '@typescript-eslint/interface-name-prefix': 2,
+  '@typescript-eslint/member-delimiter-style': [
+    2,
+    {
+      multiline: {
+        delimiter: 'comma',
+        requireLast: true,
+      },
+      singleline: {
+        delimiter: 'comma',
+        requireLast: true,
+      },
+    },
+  ],
+  '@typescript-eslint/no-angle-bracket-type-assertion': 2,
+  '@typescript-eslint/no-array-constructor': 2,
+  '@typescript-eslint/no-empty-interface': 2,
+  '@typescript-eslint/no-explicit-any': 2,
+  '@typescript-eslint/no-extra-parens': 2,
```

Comment:
no-extra-parens could probably go away, since Prettier should handle that (and it might actually conflict with parens around single parameters in function expressions)